packages/mantine/src/components/table/table-actions/TableActionItem.tsx

```diff
@@ -0,0 +1,81 @@
+import {BoxProps, CompoundStylesApiProps, Menu, PolymorphicFactory, polymorphicFactory, useProps} from '@mantine/core';
+import {ReactNode} from 'react';
+import {Button} from '../../button';
+import {useTableContext} from '../TableContext';
+
+export type TableActionItemStylesNames = 'actionItemRoot';
+
+export interface TableActionItemProps extends BoxProps, CompoundStylesApiProps<TableActionItemFactory> {
+    /**
+     * Action label
+     */
+    children: ReactNode;
+    /**
+     * Content to put on the left of the label. Usually used to render an icon
+     */
+    leftSection: ReactNode;
+    /**
+     * Whether action is primary. Primary items are visible outside the more menu
+     *
+     * @default false
+     */
+    primary?: boolean;
+
+    /**
+     * Section displayed on the right side of the label
+     */
+    rightSection?: ReactNode;
+
+    /**
+     * Disables action
+     */
+    disabled?: boolean;
+}
+
+type TableActionItemFactory = PolymorphicFactory<{
+    props: TableActionItemProps;
+    defaultRef: HTMLButtonElement;
+    defaultComponent: 'button';
+    stylesNames: TableActionItemStylesNames;
+    compound: true;
+}>;
+
+const defaultProps: Partial<TableActionItemProps> = {
+    primary: false,
+};
+
+export const TableActionItem = polymorphicFactory<TableActionItemFactory>(
+    (props: TableActionItemProps & {component?: any}, ref) => {
+        const {getStyles} = useTableContext();
+        const {classNames, className, style, styles, vars, children, primary, component, ...others} = useProps(
+            'PlasmaTableActionItem',
+            defaultProps,
+            props,
+        );
+
+        if (primary) {
+            return (
+                <Button
```

Comment:
In the case of the row actions column, it won't be a Button for primaries I think, but more an ActionIcon, and in the case of a context menu it will be perhaps something else. I wonder if we should let the parent decide which component to use (for example TableActionsHeader)